src/fsharp/NameResolution.fs

```diff
@@ -1992,13 +1996,16 @@ let CheckForTypeLegitimacyAndMultipleGenericTypeAmbiguities
 //-------------------------------------------------------------------------
 
 /// Perform name resolution for an identifier which must resolve to be a namespace or module.
-let rec ResolveLongIndentAsModuleOrNamespaceOrStaticClass sink (atMostOne: ResultCollectionSettings) amap m allowStaticClasses first fullyQualified (nenv: NameResolutionEnv) ad (id:Ident) (rest: Ident list) isOpenDecl =
+let rec ResolveLongIndentAsModuleOrNamespaceOrStaticClass (g: TcGlobals) sink (atMostOne: ResultCollectionSettings) amap m allowStaticClasses first fullyQualified (nenv: NameResolutionEnv) ad (id:Ident) (rest: Ident list) isOpenDecl =
+
+    // If the selected language version doesn't support open static classes then turn them off.
+    let allowStaticClasses = allowStaticClasses && g.langVersion.SupportsFeature LanguageFeature.OpenStaticClasses
```

Comment:
This should be: `allowStaticClasses && amap.g.landVersion.SupportsFeature LanguageFeature.OpenStaticClasses` - don't need to pass `TcGlobals` as `amap` already has it.

Reply:
Seems `ImportMap` is not a record and because of that, we need to type it, `amap: Import.ImportMap`